src/main/java/com/icthh/xm/tmf/ms/communication/service/topic/TopicMessageHandler.java

```diff
@@ -0,0 +1,37 @@
+package com.icthh.xm.tmf.ms.communication.service.topic;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.icthh.xm.commons.topic.domain.TopicConfig;
+import com.icthh.xm.commons.topic.message.MessageHandler;
+import com.icthh.xm.tmf.ms.communication.messaging.handler.TemplatedEmailMessageHandler;
+import com.icthh.xm.tmf.ms.communication.utils.ExecuteTenantContextUtils;
+import com.icthh.xm.tmf.ms.communication.web.api.model.CommunicationMessageCreate;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class TopicMessageHandler implements MessageHandler {
+
+    private static final String EMAIL_TYPE = "Email";
+
+    private final ExecuteTenantContextUtils executeTenantContextUtils;
+    private final TemplatedEmailMessageHandler emailMessageHandler;
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public void onMessage(String message, String tenant, TopicConfig topicConfig) {
+        CommunicationMessageCreate communicationMessage = toObject(message);
+        communicationMessage.setType(EMAIL_TYPE);
```

Comment:
why you override type?
it can be telegram or sms too

Reply:
Because we don't have in CommunicationMessage.class that we are using  -  "**type**" field but in CommunicationMessageCreate "type" field exists and we should fill in because type must be not null

Reply:
Removed. thanks